homeassistant/helpers/frame.py

```diff
@@ -129,15 +129,19 @@ class MissingIntegrationFrame(HomeAssistantError):
 
 def report(
     what: str,
-    exclude_integrations: set | None = None,
+    *,
+    exclude_integrations: set[str] | None = None,
     error_if_core: bool = True,
+    error_if_integration: bool = False,
     level: int = logging.WARNING,
     log_custom_component_only: bool = False,
-    error_if_integration: bool = False,
 ) -> None:
     """Report incorrect usage.
 
-    Async friendly.
+    If error_if_core is True, raise instead of log if an integration is not found
+    when unwinding the stack frame.
+    If error_if_integration is True, raise instead of log if an integration is found
+    when unwinding the stack frame.
```

Comment:
What is the purpose of these two options? When are they meant to be used?

Reply:
Side note: Didn't we have an option to only raise if the integration is a core integration and log if it's a custom integration?